### PR TITLE
Letter animation

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1365509759)

### Description of Changes Made

- Adds in a new template `motif-heading.html` which will be used for h1 and h2s with the classes of `motif-heading--one`, `motif-heading--one-b` and `motif-heading--two`.
- The template takes an optional heading level
- The template animates 2 background SVGs that are set via CSS custom properties so they can update with the theme.
- There is a 0.9 opacity set on the SVGs from which i've tested the colours against the background colour which they all pass at AA large at a minimum:
- To achieve this the first letter is wrapped in a `<span>` and to avoid issues when announcing the title via a screen reader an aria-label is added that matches the title

<details>
  <summary>Expand to see more</summary>

Screenshot 2024-01-31 at 14 41 00

Screenshot 2024-01-31 at 14 39 25

Screenshot 2024-01-31 at 14 39 17

Screenshot 2024-01-31 at 14 39 04

Screenshot 2024-01-31 at 14 35 50

Screenshot 2024-01-31 at 14 35 30

</details>

### How to Test

Across the site on a blog page e.g. http://localhost:8000/blog/ai-in-content-management-wagtail-ai-webinar/
or in the pattern library on the components page: http://localhost:8000/pattern-library/render-pattern/patterns/styleguide/components/components.html

### Screenshots

<details>
  <summary>Expand to see more</summary>

Screenshot 2024-01-31 at 14 41 43

Screenshot 2024-01-31 at 14 41 48

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [ ] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [x] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [x] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/static_src/sass/base/_typography.scss [hunk 1]
```diff
@@ -42,14 +42,17 @@ h6 {
     @include heading-text();
     font-weight: $weight--light;
 
+    // Legacy CMS used italics in headings so reset that here
+    i {
+        font-style: normal;
```
Comment (helenb) [inline tbx/static_src/sass/base/_typography.scss:47]: Should we also reset the bold here?

File: tbx/static_src/sass/components/_motif-heading.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+@use "config" as *;
+
+.motif-heading {
+    $root: &;
+
+    &__drop-cap {
+        color: transparent;
+        background-image: var(--letter-motif-two), var(--letter-motif-one);
+        background-clip: text;
+        background-color: $color--white;
```
Comment (helenb) [inline tbx/static_src/sass/components/_motif-heading.scss:10]: Could you add a comment here to explain that the background-color is essential as an accessibility fallback for any browsers that don't support background-clip? And should this also be `var(--color--heading)`?

File: tbx/static_src/sass/components/_motif-heading.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        background-repeat: no-repeat;
+        background-size: 100%;
+        -webkit-text-fill-color: transparent;
+        animation: motifAnimation 1.7s cubic-bezier(0.3, 0, 0.6, 1);
+        background-origin: border-box;
+
+        @include reduced-motion() {
+            animation: none;
```
Comment (helenb) [inline tbx/static_src/sass/components/_motif-heading.scss:18]: great stuff

File: tbx/static_src/sass/config/_variables.scss [hunk 4]
```diff
@@ -38,6 +38,11 @@ $color--mid-grey: #282e30;
...
     --color--decoration: #{$color--grey-50};
     --color--border: #{$color--grey-60};
+    // Motif heading flames
+    // - The 'fill' colour is inlined with the '#' of the hex being replaced with '%23' e.g. #ff0000 becomes %23ff0000
+    // - The hex colours used should be the same as --color--accent-two and --color--accent-three
+    --letter-motif-one: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="71" height="81" viewBox="0 0 71 81"><path d="M66.805 52.031c-.374.779-1.505.626-1.67-.22-.945-5.06-3.055-9.834-5.006-11.73l-.154.457C49.657 69.905 23.035 78.79 13.979 80.956c-.861.22-1.465-.83-.855-1.473 8.336-8.564 9.757-18.821 4.668-30.974-.988-2.251-1.957-5.061-2.673-7.786 0 0-.9-3.182-1.376-7.295l-.135.067c-5.13 4.045-7.97 9.986-9.265 13.338-.324.795-1.438.761-1.688 0C1.52 43.177-.11 35.966 1.535 29.23 5.546 12.947 20.346-.12 37.883.947c.827.051 1.11 1.05.434 1.558-2.168 1.76-4.085 4.62-5.45 8.682-3.157 9.394 2.166 27.048 2.166 27.048S49.943 27.352 53.1 17.96c1.382-4.113 1.58-7.55.914-10.258-.183-.795.647-1.421 1.287-.981 14.266 9.293 16.738 26.219 14.865 35.477-.808 4.08-2.323 7.583-3.36 9.834z" fill="%237045ff" style="opacity: 0.9"/></svg>');
```
Comment (helenb) [inline tbx/static_src/sass/config/_variables.scss:44]: I don't understand the `%23` in the fill colour - what does this do?

**Review by helenb — APPROVED** [review]
This is looking really great - top work. I have a couple of really small nitpicks.

1) in high contrast mode, the background image looks like it slightly overlaps the letter - could we just hide it?

Screenshot 2024-01-31 at 15 38 25


I also noticed the other day that the pull quote icon isn't very visible in high contrast mode with the light option, because of the white background, so maybe we could do something similar.

2) The svg looks a little pixelated, I'm not sure why - is there a better version Ben could provide maybe? Again this also applies to the pullquote icon.

Screenshot 2024-01-31 at 15 27 03


Anyway, really nice work. Love how accessible it all is - let's show this in TNV tomorrow as well!

File: tbx/static_src/sass/base/_typography.scss (hunk 1 above)
Comment (chris-lawton) [reply tbx/static_src/sass/base/_typography.scss:47]: I can add that in that will take effect in heading blocks. We'll need to set it separately for rich text headings as they don't get the `.heading` class applied

**Comment by helenb** [comment]
> 1. in high contrast mode, the background image looks like it slightly overlaps the letter - could we just hide it?

Having experimented I don't think that little blip is the background image - it appears on my screen regardless and appears to be part of the font.

File: tbx/static_src/sass/base/_typography.scss (hunk 1 above)
Comment (chris-lawton) [reply tbx/static_src/sass/base/_typography.scss:47]: Added in https://github.com/torchbox/torchbox.com/pull/80/commits/3c3f4a73f109af533e78eac65820d79aee268c44

File: tbx/static_src/sass/components/_motif-heading.scss (hunk 2 above)
Comment (chris-lawton) [reply tbx/static_src/sass/components/_motif-heading.scss]: Good shout - added in https://github.com/torchbox/torchbox.com/pull/80/commits/f0a0a4ca81dd85da11300360870942aadf162862 and updated to use the css custom property - thanks!

File: tbx/static_src/sass/config/_variables.scss (hunk 4 above)
Comment (chris-lawton) [reply tbx/static_src/sass/config/_variables.scss:44]: It's replacing the `#` you'd usually use when referencing a hex. With it the `#` interferes with the correct parsing of the data and you don't see the colours. Encoding it as `%23` ensures that it is treated as a literal character in the SVG data.

**Comment by chris-lawton** [comment]
> Having experimented I don't think that little blip is the background image - it appears on my screen regardless and appears to be part of the font.

I've noticed this as well - not sure if it's an issue with the drop cap but I don't think it's related to the SVG.

**Pull request merged**